src/api-service/__app__/onefuzzlib/orm.py

```diff
@@ -281,18 +291,56 @@ def get_keys(self) -> Tuple[KEY, KEY]:
 
         return (partition_key, row_key)
 
+    @classmethod
+    def hide_secrets(
+        cls,
+        model: BaseModel,
+        hider: Callable[["SecretData"], None],
+        visited: Set[int] = set(),
+    ) -> None:
+        model_type = type(model)
+        if id(model) in visited:
+            return
+
+        visited.add(id(model))
+
+        if _TYPE_HAS_SECRETS.get(model_type, True):
```

Comment:
I'm wondering what happens in this case:

```python
class B(BaseModel):
    c: SecretData[str]

class A(BaseModel):
    b: Optional[List[B]]
```

Consider hiding the secrets for these iteratively:
```python
A()
A(b=[B(c="hi")])
```

I think that `getattr` will return None on the first instance, which falls through the later results.  This will result in _TYPE_HAS_SECRETS falling through as False, even though the type has nested secrets.